### PR TITLE
Issue 43638: String output from JSPs

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
We've gotten stricter about always encoding Strings in JSPs

#### Changes
* Ensure String are passed through h()

### CODE REVIEW

**Review by labkey-bpatel — APPROVED** [review]

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp [hunk 1]
```diff
@@ -271,8 +271,8 @@
         <%
             if (startDate != null && endDate != null) {
         %>
-        startDateOfSchedule = new Date("<%= startDate %>");
-        endDateOfSchedule = new Date("<%= endDate %>");
+        startDateOfSchedule = new Date("<%= h(startDate) %>");
+        endDateOfSchedule = new Date("<%= h(endDate) %>");
```
Comment (labkey-adam) [inline WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp]:
```suggestion
        endDateOfSchedule = new Date("<%= q(endDate) %>");
```

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp [hunk 2]
```diff
@@ -271,8 +271,8 @@
         <%
             if (startDate != null && endDate != null) {
         %>
-        startDateOfSchedule = new Date("<%= startDate %>");
-        endDateOfSchedule = new Date("<%= endDate %>");
+        startDateOfSchedule = new Date("<%= h(startDate) %>");
```
Comment (labkey-adam) [inline WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp]:
```suggestion
        startDateOfSchedule = new Date("<%= q(startDate) %>");
```

**Review by labkey-adam — CHANGES_REQUESTED** [review]
I think `q()` is more appropriate for JavaScript parameters

File: WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp (hunk 1 above)
Comment (labkey-jeckels) [reply WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/calendars/OnCallCalendar.jsp]: Indeed, but q() handles adding the quotes on its own. Updated change pushed.

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**